### ISSUE TITLE
FEAT: Custom themes

### ISSUE THREAD

**Issue by esledov** [beekeeper-studio/beekeeper-studio#886]
Could you provide an option to change theme colors? Or provide instructions on how to change them in the code?
Are the themes codemirror-based?

Or at least one more traditional theme option, somewhere between very bright white and very deep black would be nice.

**Comment by rathboma** [comment]
Certainly! It's kind of a pain to do right now but there are light and dark theme `scss` files in the source tree.

I'm going to mark this as accepted, but this is certainly not a priority for me right now and probably won't get done until I have a way to sustain full time work on the project.

**Comment by igoracmelo** [comment]
@rathboma is it possible to change this file after build?
i have beekeeper installed and i wish to adjust the CSS/SCSS just to change some colors

**Comment by zanuka** [comment]
@rathboma I plan on submitting a PR soon for a a new Theme Manager feature that hooks in to the top menu : View > Theme and will support popular VSCode themes as well as the ability for users to upload both .json and .tmTheme theme files. Any other details you'd like to specify? I wasn't sure if you already had a spec for it, but my plan is to have the theme manager displayed in a modal component, similar to the way WorkSpaceSignInModal is displayed. This will also address the plan specified here to support common themes - https://github.com/beekeeper-studio/beekeeper-studio/issues/2836

**Comment by rathboma** [comment]
Hey @zanuka . No spec or anything, but a model sounds like the way we'd do it for sure.

Would certainly love to support both vscode and tmtheme files.

Have you done any work on this yet? From your comment it sounds like you're mostly done with it?

**Comment by zanuka** [comment]
Sounds great @rathboma, thanks for the response. Yes, I've completed an initial version that supports both vscode and thTheme files, as well as a 'preview' option so users can see what the theme will look like before selecting it. I'm working out a few kinks on the theme file upload option and should have a PR ready soon. From that point it can be iterated on as needed once you try it out :-)

**Comment by zanuka** [comment]
@rathboma I have my branch with my current idea on a theme manager implementation here if you have time to review / run / test and provide some feedback before it gets to a PR state. I'm not set on any specific design styles either and could remove the theme 'preview' cards as well, just an idea at this point. 
https://github.com/zanuka/beekeeper-studio/tree/feature/support-common-themes

I've included a few screenshots as well, to open up some discussion / feedback. Keep in mind on my current branch, I'm still addressing a few issues with the theme file uploads but the current version should remember the users selected theme after quitting/restarting the app. 

Preview mode is also functional, so users can see what it will look like before selecting. 

![Image](https://github.com/user-attachments/assets/b301bda7-c758-4938-bb50-2961b538f662)
![Image](https://github.com/user-attachments/assets/4895baed-404c-4bbe-8210-e0d72e569b9d)
![Image](https://github.com/user-attachments/assets/c6b92cc6-1016-4a5d-b18b-4cc1a8e5ea0b)
![Image](https://github.com/user-attachments/assets/b5717e0f-5365-4387-8b3b-9716afb4cf60)

**Comment by zanuka** [comment]
@rathboma did you want me to create a PR for this or did you have any other feedback / requirements

**Comment by zanuka** [comment]
happy to just submit a PR @rathboma but was standing by to see if you had a chance to review the version on my fork

**Comment by davidawad** [comment]
Found this issue after looking for the same thing. 

@zanuka your screenshots look awesome from a design standpoint, in my opinion just go for the PR, I'll help you polish it if you really need to.

**Comment by zanuka** [comment]
@davidawad ok sounds good, I'll try to get the PR created soon and we can go from there

**Comment by rathboma** [comment]
@zanuka Hey! Sorry I totally missed your post with the screenshots.

Looks awesome actually. Let me get one of the Beekeeper Studio devs involved here to help get this pushed along :-)

**Comment by rathboma** [comment]
@zanuka Please create a PR! We'll get it reviewed, updated, and hopefully merged if this all works

**Comment by zanuka** [comment]
@rathboma sounds good! I'll rebase and get PR created, hopefully today. thanks for the feedback

**Comment by zanuka** [comment]
@rathboma just created here - https://github.com/beekeeper-studio/beekeeper-studio/pull/3137
